src/action.rs

```diff
@@ -7,34 +7,37 @@ mod session;
 mod shutdown;
 mod watch;
 
-pub use drop::{DropDatabase, DropDatabaseFuture};
-pub use list_databases::{ListDatabases, ListSpecificationsFuture};
+pub use drop::{DropDatabase, DropDatabaseFuture, DropDatabaseOptions};
+pub use list_databases::{ListDatabases, ListDatabasesOptions, ListSpecificationsFuture};
 pub use perf::{WarmConnectionPool, WarmConnectionPoolFuture};
-pub use session::{StartSession, StartSessionFuture};
+pub use session::{SessionOptions, StartSession, StartSessionFuture};
 pub use shutdown::{Shutdown, ShutdownFuture};
-pub use watch::{Watch, WatchExplicitSessionFuture, WatchImplicitSessionFuture};
+pub use watch::{
+    ChangeStreamOptions,
+    FullDocumentBeforeChangeType,
+    FullDocumentType,
+    Watch,
+    WatchExplicitSessionFuture,
+    WatchImplicitSessionFuture,
+};
 
 macro_rules! option_setters {
     (
         $opt_field:ident: $opt_field_ty:ty;
-        $(
-            $(#[$($attrss:tt)*])*
-            $opt_name:ident: $opt_ty:ty,
-        )+
+        $($opt_name:ident: $opt_ty:ty,)+
     ) => {
         fn options(&mut self) -> &mut $opt_field_ty {
             self.$opt_field.get_or_insert_with(<$opt_field_ty>::default)
         }
 
-        #[cfg(test)]
-        #[allow(dead_code)]
-        pub(crate) fn with_options(mut self, value: impl Into<Option<$opt_field_ty>>) -> Self {
+        /// Set all options.  Note that this will replace all previous values set.
+        pub fn with_options(mut self, value: impl Into<Option<$opt_field_ty>>) -> Self {
             self.options = value.into();
             self
         }
 
         $(
-            $(#[$($attrss)*])*
+            #[doc = concat!("Set the [`", stringify!($opt_field_ty), "::", stringify!($opt_name), "`] option.")]
```

Comment:
With the options struct back to the place where there's canonical docs for the individual options, this means that the option_setters invocation don't need to duplicate the docs.